### ISSUE TITLE
aud should be checked for equality

### ISSUE THREAD

**Issue by sgsfak** [curityio/flask-of-oil#12]
At line [37](https://github.com/curityio/flask-of-oil/blob/91daf66ec55a2a433ed4da02b07e8ea95b94bcda/flask_of_oil/jwt_validator.py#L67) of jwt_validator.py I read:

```
        if self.aud not in aud:
            self.logger.debug("Invalid audience %s, expected %s" % (aud, self.aud))
            return {"active": False}
```

which actually performs a prefix/suffix/contains match on the Audience information (assuming that we have a single Audience string specified)

So with the following setup:
```
_oauth.configure_with_jwt("https://example.com/certs", "https://example.com", "TestAudience")
```

any token with `aud` that contains TestAudience (e.g. MyTestAudience or AnotherTestAudience2) is allowed.

**Comment by mtrojanowski** [comment]
Hey @sgsfak,

Good catch, thanks for letting us know! Indeed there is a bug when validating the `aud` claim. I've proposed a fix in #13 but it's not checking for equality. The JWT spec allows the `aud` claim to be an array, so I'm instead casting the claim to an array if it's not one.

**Comment by mtrojanowski** [comment]
The fix is now merged and released in version 1.2.0. Thanks again for reporting this.
